### PR TITLE
Spotify auth and playlist stuff

### PR DESCRIPTION
**Pull request by ryanhu021**

### CODE REVIEW

File: packages/backend/src/routes/playlist-routes.ts [hunk 1]
```diff
@@ -1,69 +1,93 @@
...
+
+    // create playlist
+    const playlist = new Playlist({
+      name,
+      creator,
+      creatorName,
+      songs: [],
+      imageUrl,
+    });
     const newPlaylist = await playlist.save();
```
Comment (ktaschek) [inline packages/backend/src/routes/playlist-routes.ts:46]: should only save playlist after checking if use exists

**Review by ktaschek — CHANGES_REQUESTED** [review]
only thing was that I wasn't sure if we needed to save the playlist in the db before checking if the user existed

File: packages/backend/src/routes/song-routes.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import express from "express";
 import { deezerUrlSearch } from "../services/deezer-services";
-import { spotifySongFetch } from "../services/spotify-services";
+import { spotifyAuthUrl, spotifySongFetch } from "../services/spotify-services";
 import { Song } from "../models/song-model";
 import { Track } from "../services/spotify-services";
+import { AuthRequest, auth } from "../util/auth";
```
Comment (ktaschek) [inline packages/backend/src/routes/song-routes.ts:6]: authrequest and auth never used

File: packages/backend/src/routes/playlist-routes.ts (hunk 1 above)
Comment (ryanhu021) [reply packages/backend/src/routes/playlist-routes.ts:46]: it already checks for that in the auth middleware function

**Pull request merged**